api/v1alpha1/endpointmonitor_types.go

```diff
@@ -288,6 +292,70 @@ type PingdomConfig struct {
 	PostDataEnvVar string `json:"postDataEnvVar,omitempty"`
 }
 
+// PingdomTransactionConfig defines the configuration for Pingdom Transaction Monitor Provider
+type PingdomTransactionConfig struct {
+
+	// Check status: active or inactive
+	// +optional
+	Paused bool `json:"paused,omitempty"`
+
+	// Custom message that is part of the email and webhook alerts
+	// +optional
+	CustomMessage string `json:"custom_message,omitempty"`
+
+	// TMS test intervals in minutes. Allowed intervals: 5,10,20,60,720,1440. The interval you're allowed to set may vary depending on your current plan.
+	// +optional
+	// +kubebuilder:validation:Enum=5;10;20;60;720;1440
+	Interval int `json:"interval,omitempty"`
+
+	// Name of the region where the check is executed. Supported regions: us-east, us-west, eu, au
+	// +optional
+	// +kubebuilder:validation:Enum=us-east;us-west;eu;au
+	Region string `json:"region,omitempty"`
+
+	// Send notification when down X times
+	SendNotificationWhenDown int64 `json:"send_notification_when_down,omitempty"`
+
+	// Check importance- how important are the alerts when the check fails. Allowed values: low, high
+	// +optional
+	// +kubebuilder:validation:Enum=low;high
+	SeverityLevel string `json:"severity_level,omitempty"`
+
+	// steps to be executed as part of the check
+	// +required
+	Steps []PingdomStep `json:"steps"`
+
+	// List of tags for a check. The tag name may contain the characters 'A-Z', 'a-z', '0-9', '_' and '-'. The maximum length of a tag is 64 characters.
+	Tags []string `json:"tags,omitempty"`
+
+	// `-` separated set list of integrations ids (e.g. "91166-12168")
+	// +optional
+	AlertIntegrations string `json:"alertIntegrations,omitempty"`
```

Comment:
I personally would prefer a string array here because it's more natural for yaml and users. On the other hand, pingdom user might be used to existing format, What is your opinion?

Reply:
i myself prefer arrays as well for such conditions, pushed a change recently with array for grafana cloud

Reply:
Unfortunately this requires a refactoring. The problem is the alertIntegration, alertContacts and TeamAlerts are defined in Provider struct 
https://github.com/stakater/IngressMonitorController/blob/88a3c4704511084a0d44f43ff9e47e6a603eb331/pkg/config/config.go#L65-L67
This is used by all provider implementation. Probably string was used to allow more flexible setup and let the parsing upto the specific provider implementation. 
I think its fine if we keep string here, also to align to existing pingdom configuration. I would like to do changes here